File: Packages/module-ai/Sources/ModuleAI/Application/UseCases/AIChat/SendLikeFeedbackUseCase.swift
//
//  SendLikeFeedbackUseCase.swift
//  ModuleAI
//
//  Created by Codex on 2026/1/8.
//

import Foundation

struct SendLikeFeedbackUseCase {
    private let repository: AIChatRepository

    init(repository: AIChatRepository) {
        self.repository = repository
    }

    func execute(historyDetailId: Int, like: String) async throws {
        try await repository.sendLikeFeedback(historyDetailId: historyDetailId, like: like)
    }
}

File: app-android/feature/ai-chat/src/main/java/com/aidatainsight/android/feature/aichat/application/usecase/SendLikeFeedbackUseCase.kt
package com.aidatainsight.android.feature.aichat.application.usecase

import com.aidatainsight.android.feature.aichat.domain.AIChatRepository

class SendLikeFeedbackUseCase(
    private val repository: AIChatRepository,
) {
    suspend operator fun invoke(historyDetailId: Int, like: String): Result<Unit> {
        return runCatching {
            repository.sendLikeFeedback(historyDetailId = historyDetailId, like = like)
        }
    }
}
